app/models/admin/proposals/proposal.rb

```diff
@@ -68,29 +68,31 @@ def labels(pull_right)
 
     labels << case successful
               when true
-                generate_label(:success, 'Successful', pull_right)
+                generate_label(:success, 'Successful')
               when false
-                generate_label(:danger, 'Unsuccessful', pull_right)
+                generate_label(:danger, 'Unsuccessful')
               else
                 case approved
                 when true
-                  generate_label(:success, 'Approved', pull_right)
+                  generate_label(:success, 'Approved')
                 when false
-                  generate_label(:danger, 'Rejected', pull_right)
+                  generate_label(:danger, 'Rejected')
                 else
-                  generate_label(:warning, 'Waiting for Approval', pull_right)
+                  generate_label(:warning, 'Waiting for Approval')
                 end
               end
 
-    labels << generate_label(:danger, 'Late', pull_right) if late
-    labels << generate_label(:danger, 'Has Debtors', pull_right) if has_debtors
+    labels << generate_label(:danger, 'Late') if late
+    labels << generate_label(:danger, 'Has Debtors') if has_debtors
 
+    labels_html = labels.join("\n").html_safe
+
+    # Wrap the whole list of labels in a float right so that the margins stay preserved.
     if pull_right
```

Comment:
This bit of code is specific to the labels on proposals. It is once shown float-right (in the index) and once without float-right (on the proposal show page). We could just wrap it in a the same div at it's single occurence I suppose.

Wrapping generated labels in float-rights directly (the old way) is not done anywhere else, but it's a small addition and it might come up in the future. I am happy to take it out but it feels like something that could be useful and does not take a lot of effort to maintain other than replacing it with float-end once we upgrade to Bootstrap 5 on the admin site.